### DISCUSSION TITLE
BackgroundTasks where one task raises an Exception

### DISCUSSION THREAD

**Discussion by cancan101** [encode/starlette#1584] [Ideas] [answered]
This is somewhat a request for documentation and somewhat of a suggestion.

Right now if there are more than one task added to a `BackgroundTasks` and then when the tasks are run it raises an Exception, subsequent tasks are not run. This is likely unexpected where the developer would expect the remaining tasks to run (especially given the ordering might be arbitrary).
https://github.com/encode/starlette/blob/165592fb89ad26383b4a8122b4829bdfe4c64d60/starlette/background.py#L41-L43

this doesn't seem to be documented here: https://www.starlette.io/background/#backgroundtasks

I suggest at the very least documenting this and possibly having a flag to catch error and then continue with the other tasks, perhaps then throwing a combined error set at the end.

**Comment by Kludex** [comment, marked as answer]
PR to improve documentation regarding this behavior is welcome. PR with a test presenting the current behavior is also welcome.

I have the feeling that no one asked about this before because the use-case of most is a single task. 🤔

Thanks for open this discussion @cancan101 ! :)

**Reply by Kludex** [reply]
This is already documented on the https://www.starlette.io/background/. :pray:
